Bottom: 436f5ab6c274613a918363f65c94d6f80dbd85b5
Top:    2fed6d12d66d6d88221f56e27586102007b3f294
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-02-02 22:26:26 +0000

Add profiling to file load











---

diff --git a/src/file.c b/src/file.c
index fb1c923..fb75652 100644
--- a/src/file.c
+++ b/src/file.c
@@ -350,6 +350,10 @@ LoadPCB (char *Filename)
 {
   PCBTypePtr newPCB = CreateNewPCB (False);
   Boolean units_mm;
+  clock_t start, end;
+  double elapsed;
+
+  start = clock();
 
   /* new data isn't added to the undo list */
   if (!ParsePCB (newPCB, Filename))
@@ -380,6 +384,10 @@ LoadPCB (char *Filename)
 	  CreateDefaultFont ();
 	}
 
+  end = clock();
+  elapsed = ((double) (end - start)) / CLOCKS_PER_SEC;
+  printf ("Loading file %s took %f\n", Filename, elapsed);
+
       /* clear 'changed flag' */
       SetChangedFlag (False);
       PCB->Filename = MyStrdup (Filename, "LoadPCB()");
